### PR TITLE
Fix runtime-id and system.pid tags not being set on distributed traces

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**
# Bug fix

## Description

The runtime id tag was not being set on distributed traces.

## Fix

Set the runtime id tag for spans that don't have strong parent references.

# Checklist
- [x] Regression test added; or
- [x] Description of manual testing performed and explanation is included in the code and/or PR.
- [x] Added to the correct milestone.

### CODE REVIEW

File: ddtrace/tracer.py [hunk 1]
```diff
@@ -581,6 +580,9 @@ def _start_span(
                 # We must always mark the span as sampled so it is forwarded to the agent
                 span.sampled = True
 
+        if not span._parent:
+            span.meta["runtime-id"] = get_runtime_id()
```
Comment (brettlangdon) [inline ddtrace/tracer.py:584]: I wonder if this should go into `TraceTagsProcessor` instead?

Comment (brettlangdon) [reply ddtrace/tracer.py:584]: unless, does profiling need to access this at all? I know we need to use the same `runtime-id`, but profiling doesn't read it from the span does it?

Comment (Kyle-Verhoog) [reply ddtrace/tracer.py:584]: profiling uses the `runtime` API to get the runtime id. 

> I wonder if this should go into TraceTagsProcessor instead?

Hmm yeah this is doable. A SpanProcessor would probably make more sense. I think most of the tags set in `start_span` could be moved out to a `DefaultTagsSpanProcessor` or similar.

I'm ok with merging this as-is and making the refactor after WDYT @brettlangdon?

Comment (brettlangdon) [reply ddtrace/tracer.py:584]: well, the question is whether it should be put on trace root, chunk root, or all spans?

I think chunk root makes the most sense right now, at least until we have a more optimized way to associate a tag with all spans.

but I am ok with taking this as-is, and then follow-up moves to chunk root or all spans.

**Review by brettlangdon — APPROVED** [review]

File: ddtrace/tracer.py (hunk 1 above)
Comment (Kyle-Verhoog) [reply ddtrace/tracer.py:584]: I would think that it's like service where all spans under a span with a defined runtime id will share the runtime id. We should check with the backend teams to see what it looks for.

File: ddtrace/tracer.py [hunk 2]
```diff
@@ -551,7 +551,6 @@ def _start_span(
             )
             span._local_root = span
             span.metrics[system.PID] = self._pid or getpid()
-            span.meta["runtime-id"] = get_runtime_id()
```
Comment (Yun-Kim) [inline ddtrace/tracer.py:554]: Do we not need to set `runtime-id` on a span if it's not a distributed part of a distributed trace?

**Review by Yun-Kim — COMMENTED** [review]
Just one question 😄

File: ddtrace/tracer.py (hunk 1 above)
Comment (brettlangdon) [reply ddtrace/tracer.py:584]: yeah, we can check.

File: ddtrace/tracer.py (hunk 2 above)
Comment (brettlangdon) [reply ddtrace/tracer.py:554]: @Kyle-Verhoog looking at this, do we need to do the same for `system.PID` as well?

File: releasenotes/notes/runtimeid-39dfdf1e0c1af45d.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+---
+fixes:
+  - |
+    Fix runtime id tag not being set on distributed traces.
```
Comment (brettlangdon) [inline releasenotes/notes/runtimeid-39dfdf1e0c1af45d.yaml:4]:
```suggestion
    Fix runtime-id and system.pid tags not being set on distributed traces.
```

**Review by brettlangdon — DISMISSED** [review]

File: ddtrace/tracer.py (hunk 2 above)
Comment (Kyle-Verhoog) [reply ddtrace/tracer.py:554]: We do still need to @Yun-Kim. In general we want root spans to have the runtime-id and os pid.

Comment (Yun-Kim) [reply ddtrace/tracer.py:554]: I thought in distributed tracing the new root span technically should still have a parent id right? Or is there not a strong parent reference in a distributed tracing context?

Comment (Kyle-Verhoog) [reply ddtrace/tracer.py:554]: Correct, it has a parent id and not a strong parent reference.

Using `_parent` isn't _great_ but it's the best we can do I think. It does mean that we'll report the runtime id and os pid for intra-process execution crossing in the local root span (like the local root of a new thread).

**Review by brettlangdon — APPROVED** [review]

**Review by Yun-Kim — APPROVED** [review]

**Pull request merged**